src/engine/input/key-mapping/InputEventItem.cs

```diff
@@ -0,0 +1,138 @@
+﻿using System;
+using Godot;
+
+public class InputEventItem : Node
+{
+    [Export]
+    public NodePath ButtonPath;
+
+    internal InputActionItem AssociatedAction;
+
+    internal bool JustAdded;
+
+    private Button button;
+
+    private bool waitingForInput;
+
+    public InputEventWithModifiers AssociatedEvent { get; set; }
```

Comment:
I don't feel particularly good about storing the Godot input events? Which inherits from Resource, so this is highly dubious to me. If this information truly needs to be stored, we need a custom class that grabs the needed info from an InputEventWithModifiers.

Reply:
storing the InputEventWithModifiers makes the modification of the InputMap much easier.
To save just the needed information I have created a jsonConverter

Reply:
Reading this: `<para>Unlike <see cref="T:Godot.Object" />s, References keep an internal reference counter so that they are automatically released when no longer in use, and only then. References therefore do not need to be freed manually with <see cref="M:Godot.Object.Free" />.` bit of documentation on Reference, makes it a bit more stomachable for me to use InputEventWithModifiers in this manner.

Reply:
It states that i do not need to free them manually, but what if I do it anyways?
Shouldn't some calls to Free resolve any memory leaks without any major code reworks?

Reply:
The Godot input tutorials don't call Free on the input events passed to scripts. I'm not sure if it causes any problems to call Free manually. At least with deleting Nodes you run the risk of getting object disposed exceptions when you don't want to get them.

Reply:
The Free method exists, but it prints
`Can't 'free' a reference.`
To the console.
Calling dispose() gives me ObjectDisposedExceptions...

Reply:
Then you should follow the documentation and not call free.
The disposed exceptions indicate to me that you are disposing the thing at inopportune times.
This is why I'd really prefer our own class that behaves exactly as needed for the input key binding tracking, even if that means some extra code to convert between that and the godot types.